Pointers/UniquePtr.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <iostream>
+
+using std::cout;
+using std::endl;
+
+template<typename T>
+class UniquePtr {
+public:
+	UniquePtr(T* ptr);
+	~UniquePtr();
+	// Don't allow copy
+	UniquePtr(const UniquePtr<T>& other) = delete;
+	UniquePtr<T>& operator=(const UniquePtr<T>& other) = delete;
+	// Allow move
+	UniquePtr(UniquePtr<T>&& other) noexcept;
+	UniquePtr<T>& operator=(UniquePtr<T>&& other) noexcept;
+
+	T& operator*() const;
+	T* operator->() const;
+private:
+	T* m_ptr;
+};
+
+template<typename T>
```

Comment:
When we write a templated class we will put the implementation in a separate inl file and the header file will include it
This way we get the logical separation between the interface and its implementation
The header includes it because templates must be resolved at compile time